support/elasticsearch_search.erl

```diff
@@ -339,7 +339,7 @@ map_must_not({cat_exclude, Name}, Context) ->
         Filtered ->
             {true, [{terms, [{category, Filtered}]}]}
     end;
-map_must_not({filter, [Key, Operator, Value]}, Context) when is_list(Key) ->
+map_must_not({filter, [Key, Operator, Value]}, Context) when is_list(Key) and not is_list(Operator) ->
```

Comment:
Is this change necessary?

Reply:
Yes, this was necessary because this filter:

{filter, [[<<"value1">>, exists], [<<"value2">>, exists], [<<"value3">>, exists]]}

Would match the map_must_not while it should go to the map_must.

Reply:
I see. But how does your change help?

```
Key = [<<"value1">>, exists]
Operator = [<<"value2">>, exists]
Value = [<<"value3">>, exists]]}
```

So `Operator` is still a list and wouldn’t match the guard.

Reply:
I see. Can you change it to `when is_list(Key), is_atom(Operator)` instead? It’s slightly simpler and better intelligible.

Reply:
Done